include/ozo/binary_deserialization.h

```diff
@@ -232,4 +232,23 @@ basic_result<T>& recv_result(basic_result<T>& in, const oid_map_t<M>&, basic_res
     return out;
 }
 
+template <typename T, typename M>
+basic_result<T>& recv_result(basic_result<T>& in, const oid_map_t<M>& oid_map, std::reference_wrapper<basic_result<T>> out) {
+    return recv_result(in, oid_map, out.get());
+}
+
+template <typename M>
+basic_result<shared_native_result_handle>& recv_result(basic_result<native_result_handle>& in,
+        const oid_map_t<M>&, basic_result<shared_native_result_handle>& out) {
+    out = basic_result<shared_native_result_handle>(
+        shared_native_result_handle(in.handle().release(), std::default_delete<PGresult> {})
+    );
+    return out;
+}
+
+template <typename T, typename M>
+no_result& recv_result(basic_result<T>&, const oid_map_t<M>&, no_result& out) {
```

Comment:
The best choice is to provide execute() method - in this case, no extra recv_result() is needed.

<details><summary><b>Something like this based on request</b></summary><p>

```c++
template <typename Query, typename Out, typename Handler>
inline auto make_async_execute_op(Query&& query, Handler&& handler) {
    auto do_nothing = [](auto, auto){};
    return impl::async_request_op<decltype(do_nothing),
            std::decay_t<Query>, std::decay_t<Handler>>{
        do_nothing,
        std::forward<Query>(query),
        std::forward<Handler>(handler)
    };
}

template <typename P, typename Q, typename Out, typename Handler>
inline void async_execute(P&& provider, Q&& query, Handler&& handler) {
    static_assert(ConnectionProvider<P>, "is not a ConnectionProvider");
    static_assert(Query<Q> || QueryBuilder<Q>, "is neither Query nor QueryBuilder");
    async_get_connection(std::forward<P>(provider),
        impl:: make_async_execute_op(
            std::forward<Q>(query),
            std::forward<Handler>(handler)
        )
    );
}
```

</p></details>